frontend/src/components/common/ToastMessage.vue

```diff
@@ -27,42 +22,18 @@ const onError = (error: any, info: string) => {
     if (axios.isAxiosError(error)) {
         const err = error;
         const axiosResponse = err.response;
-        const status = axiosResponse?.status;
-
-        const e401_authenticationErrorMsg = {
-            title: 'Error',
-            text: 'You are not logged in. Please log in.',
-        };
```

Comment:
Hi Nick, I think we still want to keep these the 401 and 403 error handling. We want the hardcode error messages, cause not all messages from backend are intended for frontend. So let's still keep these if conditions. But for the special case "can't grant self access", which is also a 403 error, we can check if the error detail code is "self_grant_prohibited", then we display the error description from the server. For the rest 403 error cases, we can still show the hardcoded message as previous. 
```
detail: {code: "self_grant_prohibited", description: "Granting roles to self is not allowed"}
```

I'll create a follow up ticket for further discussion about how server should return the error for frontend to handle. But for now, let's still follow the old way, but add the special case for  "can't grant self access". For 409 error, we don't need to hardcode, just show the error from the server.

Reply:
Bringing it back and adjusting for `self_grant_prohibited` case.